akka-typed/src/main/scala/akka/typed/scaladsl/Actor.scala

```diff
@@ -0,0 +1,399 @@
+/**
+ * Copyright (C) 2017 Lightbend Inc. <http://www.lightbend.com/>
+ */
+package akka.typed
+package scaladsl
+
+import akka.util.LineNumbers
+import scala.reflect.ClassTag
+import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.ExecutionContextExecutor
+import scala.deprecatedInheritance
+import akka.typed.{ ActorContext ⇒ AC }
+
+/**
+ * An Actor is given by the combination of a [[Behavior]] and a context in
+ * which this behavior is executed. As per the Actor Model an Actor can perform
+ * the following actions when processing a message:
+ *
+ *  - send a finite number of messages to other Actors it knows
+ *  - create a finite number of Actors
+ *  - designate the behavior for the next message
+ *
+ * In Akka the first capability is accessed by using the `!` or `tell` method
+ * on an [[ActorRef]], the second is provided by [[ActorContext#spawn]]
+ * and the third is implicit in the signature of [[Behavior]] in that the next
+ * behavior is always returned from the message processing logic.
+ *
+ * An `ActorContext` in addition provides access to the Actor’s own identity (“`self`”),
+ * the [[ActorSystem]] it is part of, methods for querying the list of child Actors it
+ * created, access to [[Terminated DeathWatch]] and timed message scheduling.
+ */
+trait ActorContext[T] { this: akka.typed.javadsl.ActorContext[T] ⇒
```

Comment:
Users must not implement this trait—I’ll add the annotation as well—so I think we’re fine because our own implementation in `akka.typed.internal` will always be in sync. But if a Scala/MiMa expert sees something I’m missing then we’re toast.